### PR TITLE
Improve Ballerina File API to use java nio

### PR DESCRIPTION
**Pull request by Arshardh**

### CODE REVIEW

**Comment by Arshardh** [comment]
The discussed changes have been made.

File: modules/ballerina-native/src/main/java/org/ballerinalang/nativeimpl/lang/files/Close.java [hunk 1]
```diff
@@ -55,10 +54,8 @@
     @Override
     public BValue[] execute(Context context) {
         BStruct struct = (BStruct) getRefArgument(context, 0);
-        BufferedInputStream is = (BufferedInputStream) struct.getNativeData("inStream");
-        BufferedOutputStream os = (BufferedOutputStream) struct.getNativeData("outStream");
-        closeQuietly(is);
-        closeQuietly(os);
+        SeekableByteChannel channel = (SeekableByteChannel) struct.getNativeData("channel");
+        closeQuietly(channel);
```
Comment (sameera-jayasoma) [inline modules/ballerina-native/src/main/java/org/ballerinalang/nativeimpl/lang/files/Close.java]: Why do we close quietly?  We need to throw an error here.

Comment (Arshardh) [inline modules/ballerina-native/src/main/java/org/ballerinalang/nativeimpl/lang/files/Close.java]: Fixed this

File: modules/ballerina-native/src/main/java/org/ballerinalang/nativeimpl/lang/files/Close.java [hunk 2]
```diff
@@ -55,20 +54,14 @@
...
+        SeekableByteChannel channel = (SeekableByteChannel) struct.getNativeData("channel");
         try {
-            if (resource != null) {
-                resource.close();
+            if (channel != null) {
+                channel.close();
             }
         } catch (IOException e) {
-            log.error("Exception during Resource.close()", e);
+            throw new BallerinaException("failed to close channel: " + e.getMessage(), e);
```
Comment (sameera-jayasoma) [inline modules/ballerina-native/src/main/java/org/ballerinalang/nativeimpl/lang/files/Close.java]: The word channel is not useful for the ballerina developers IMV

Comment (Arshardh) [inline modules/ballerina-native/src/main/java/org/ballerinalang/nativeimpl/lang/files/Close.java]: Fixed this

**Comment by shafreenAnfar** [comment]
Hi,

Unfortunately, this PR is no longer valid as we have done a major revamp in File IO architecture. Following PR [1] incorporates the related changes. Hence, I am closing this PR.

[1] https://github.com/ballerinalang/ballerina/pull/3755

**Pull request closed without merging**